modules/data_release/jsx/uploadFileForm.js

```diff
@@ -0,0 +1,218 @@
+            {this.state.error}
+          </strong>
+        </div>
+      );
+    }
+
+    return (
+      <FormElement
+        name='uploadfile'
+        fileUpload={true}
+        onSubmit={this.validateAndSubmit}>
+        <FileElement
+          name='file'
+          label='File to upload'
+          onUserInput={this.updateFormElement}
+          required={true}
+          value={this.state.formData.file}
+        />
+        <TextboxElement
+          name='version'
+          label='Version'
+          onUserInput={this.updateFormElement}
+          required={false}
+          value={this.state.formData.version}
+        />
+        <ButtonElement label='Upload File'/>
+        <div className='row'>
+          <div className='col-sm-9 col-sm-offset-3'>
+            <ProgressBar value={this.state.uploadProgress}/>
+          </div>
+        </div>
+      </FormElement>
+    );
+  }
+
+  /**
+   * Update a form Element
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding element
+   */
+  updateFormElement(formElement, value) {
+    let formData = this.state.formData;
+
+    if (value === '') {
+      formData[formElement] = null;
+    } else {
+      formData[formElement] = value;
+    }
+
+    this.setState({formData: formData});
+  }
+
+  /**
+   * Validate and submit the upload
+   */
+  validateAndSubmit() {
+    let files = this.state.data.files ? this.state.data.files : [];
+    let formData = this.state.formData;
+
+    let errorMessage = {
+      Filename: undefined,
+    };
+
+    let hasError = {
+      Filename: undefined,
+    };
+
+    if (!formData.file) {
+      errorMessage.Filename = 'You must select a file to upload';
+      hasError.Filename = true;
+      this.setState({errorMessage, hasError});
+      return;
+    }
+
+    // Grep the uploaded file name
+    let fileName = formData.file ? formData.file.name.replace(/\s+/g, '_') : null;
+
+    // Check for duplicate file names
+    let isDuplicate = files.indexOf(fileName);
+    if (isDuplicate >= 0) {
+      swal({
+        title: 'Are you sure?',
+        text: 'A file with this name already exists!\n Would you like to override existing file?',
+        type: 'warning',
+        showCancelButton: true,
+        confirmButtonText: 'Yes, I am sure!',
+        cancelButtonText: 'No, cancel it!',
+      }, function(isConfirm) {
+        if (isConfirm) {
+          this.uploadFile();
+        } else {
+          swal('Cancelled', 'Your imaginary file is safe :)', 'error');
+        }
+      }.bind(this));
+    } else {
+      this.uploadFile();
+    }
+  }
+
+  /**
+   * Upload the file to the server
+   */
+  uploadFile() {
+    let formData = this.state.formData;
+    let formObj = new FormData();
+    for (let key in formData) {
+      if (formData[key] !== '') {
+        formObj.append(key, formData[key]);
+      }
+    }
+
+    // ajax call to upload the file
+    $.ajax({
```

Comment:
Can this use the fetch API instead of jQuery?

Reply:
Is there an example somewhere that use the fetch API? Not sure how to do that...

Reply:
There are other places in this PR that you used it.. ie. modules/data_release/jsx/addPermissionForm.js line 35.

For POST requests, I know @HenriRabalais looked up how to do it a while ago but I can't remember in which module.. @HenriRabalais do you have an example to point @cmadjar to?

Reply:
@cmadjar  The media module is using fetch api https://github.com/aces/Loris/blob/master/modules/media/jsx/mediaIndex.js

Reply:
@HenriRabalais Thank you! I will look into it. :)